### ISSUE TITLE
Segmentation fault when clicking on settings

### ISSUE THREAD

**Issue by D3SOX** [jmanc3/winbar#45]
When clicking on settings in the start menu, I get a segfault. Sometimes the bar disappears and reappears and after clicking it a second or third time I get the segfault

```
[Detaching after vfork from child process 480633]
[Detaching after vfork from child process 480634]

Thread 43 "winbar" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffee5c8640 (LWP 480632)]
0x00007ffff6a539a4 in __strcpy_avx2 () from /usr/lib/libc.so.6
(gdb) backtrace
#0  0x00007ffff6a539a4 in __strcpy_avx2 () at /usr/lib/libc.so.6
#1  0x0000555555647b1a in search_icons(std::vector<IconTarget, std::allocator<IconTarget> >&) (targets=std::vector of length 166, capacity 256 = {...})
    at /home/nico/building_stuff/winbar/lib/icons.cpp:483
#2  0x000055555559da63 in paint_desktop_files() () at /home/nico/building_stuff/winbar/src/app_menu.cpp:1070
#3  0x00005555555aa31f in std::__invoke_impl<void, void (*)()>(std::__invoke_other, void (*&&)()) (__f=@0x55555599ed98: 0x55555559d8b4 <paint_desktop_files()>)
    at /usr/include/c++/11.2.0/bits/invoke.h:61
#4  0x00005555555aa2d7 in std::__invoke<void (*)()>(void (*&&)()) (__fn=@0x55555599ed98: 0x55555559d8b4 <paint_desktop_files()>)
    at /usr/include/c++/11.2.0/bits/invoke.h:96
#5  0x00005555555aa284 in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x55555599ed98)
    at /usr/include/c++/11.2.0/bits/std_thread.h:253
#6  0x00005555555aa258 in std::thread::_Invoker<std::tuple<void (*)()> >::operator()() (this=0x55555599ed98) at /usr/include/c++/11.2.0/bits/std_thread.h:260
#7  0x00005555555aa23c in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run() (this=0x55555599ed90)
    at /usr/include/c++/11.2.0/bits/std_thread.h:211
#8  0x00007ffff6bc94d4 in std::execute_native_thread_routine(void*) (__p=0x55555599ed90) at /usr/src/debug/gcc/libstdc++-v3/src/c++11/thread.cc:82
#9  0x00007ffff695c5c2 in start_thread () at /usr/lib/libc.so.6
#10 0x00007ffff69e1584 in clone () at /usr/lib/libc.so.6
```

Let me know if you need any additional information

**Comment by jmanc3** [comment]
I mentioned it in another issue but, for right now, power shuts down winbar and settings restarts it. That's on purpose. The segfault obviously isn't so I'll fix that but the restarting is.

**Comment by jmanc3** [comment]
@D3SOX shouldn't segfault when restarting anymore, as of: [6e5ce19](https://github.com/jmanc3/winbar/commit/6e5ce194522219deee08a56f4ed656e2dec04703).
Let me know.

**Comment by D3SOX** [comment]
Still crashes for me. The backtrace is slightly different:

```
Thread 5 "winbar" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffee5c8640 (LWP 495160)]
0x00007ffff6a539a4 in __strcpy_avx2 () from /usr/lib/libc.so.6
(gdb) backtrace
#0  0x00007ffff6a539a4 in __strcpy_avx2 () at /usr/lib/libc.so.6
#1  0x0000555555647c40 in search_icons(std::vector<IconTarget, std::allocator<IconTarget> >&) (targets=std::vector of length 166, capacity 256 = {...})
    at /home/nico/building_stuff/winbar/lib/icons.cpp:483
#2  0x000055555559db51 in paint_desktop_files() () at /home/nico/building_stuff/winbar/src/app_menu.cpp:1074
#3  0x00005555555aa501 in std::__invoke_impl<void, void (*)()>(std::__invoke_other, void (*&&)()) (__f=@0x555555804178: 0x55555559d973 <paint_desktop_files()>)
    at /usr/include/c++/11.2.0/bits/invoke.h:61
#4  0x00005555555aa4b9 in std::__invoke<void (*)()>(void (*&&)()) (__fn=@0x555555804178: 0x55555559d973 <paint_desktop_files()>)
    at /usr/include/c++/11.2.0/bits/invoke.h:96
#5  0x00005555555aa466 in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x555555804178)
    at /usr/include/c++/11.2.0/bits/std_thread.h:253
#6  0x00005555555aa43a in std::thread::_Invoker<std::tuple<void (*)()> >::operator()() (this=0x555555804178) at /usr/include/c++/11.2.0/bits/std_thread.h:260
#7  0x00005555555aa41e in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run() (this=0x555555804170)
    at /usr/include/c++/11.2.0/bits/std_thread.h:211
#8  0x00007ffff6bc94d4 in std::execute_native_thread_routine(void*) (__p=0x555555804170) at /usr/src/debug/gcc/libstdc++-v3/src/c++11/thread.cc:82
#9  0x00007ffff695c5c2 in start_thread () at /usr/lib/libc.so.6
#10 0x00007ffff69e1584 in clone () at /usr/lib/libc.so.6
```

**Comment by jmanc3** [comment]
@D3SOX Oops. I didn't actually lock on the mutex. Fixed in: d3c815a3973e30015938d4e56634454c1b640b0a Let me know
